test/factories/referral_factory.ex

```diff
@@ -0,0 +1,12 @@
+defmodule Safira.ReferralFactory do
+  defmacro __using__(_opts) do
+    quote do
+      def referral_factory do
+        %Safira.Contest.Referral{
+          available: true,
+          badge: build(:badge)
+        }
+      end
+    end
+  end
+end
```

Comment:
Can you move this to `test/factories/contest_factory.ex`? 🙏🏼 
We are trying to create a factories file per context.